.github/workflows/ft_tests.yml

```diff
@@ -0,0 +1,40 @@
+name: Free Threading Tests
+on: [push, pull_request]
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install uv and set the python version
+        uses: astral-sh/setup-uv@v5
+        with:
+          version: "0.7.2"
+          python-version: 3.13t
+          enable-cache: false
+      - name: Set up Python
+        run: uv python install
+      - name: Install Dependencies
+        run: |
+          set -x
+          set -e
+          uv pip install packaging setuptools pyflakes pytest-cov pytest-flakes pytest-timeout pytest-durations
+          # Test dep, consider skipping
+          uv pip install sympy hypothesis
+          uv pip install --pre --upgrade -i https://pypi.anaconda.org/scientific-python-nightly-wheels/simple cython numpy
+          uv pip install pytest-run-parallel
+      - name: Disable Coverage # see gh-198
+        run: |
+          sed -i '/plugins = Cython.Coverage/d' .coveragerc
+          sed -i 's/addopts/\#addopts/' pytest.ini
+      - name: In-place build
+        run: |
+          uv pip install . --no-build-isolation -v
+          uv run --no-project python setup.py clean
+          uv run --no-project  python setup.py build_ext --inplace
+      - name: Run Tests
+        run: |
+          set -x
+          set -e
+          uv run --no-project python -We:invalid -We::SyntaxWarning -m compileall -f -q ndindex/
+          uv run --no-project python -Xgil=0 -m pytest --parallel-threads=2 --iterations=2 -v -s \
```

Comment:
Remove `-Xgil=0`? All extensions of `ndindex` itself and its dependencies (only `numpy` has compiled extensions) should be safe already.

Reply:
Left a comment since it seems to be recommended on the [Cython docs](https://docs.cython.org/en/latest/src/userguide/freethreading.html#:~:text=If%20you%20want,useful%20for%20testing.)